skyportal/schema.py

```diff
@@ -76,21 +76,39 @@ def setup_schema():
                     "Model class ends with 'Schema'"
                 )
 
-            class Meta(object):
-                model = class_
-                sqla_session = _DBSession
+            def schema_class_factory(name):
+                return type(name, (), {'model': class_, 'sqla_session': _DBSession,
+                                       'ordered': True, 'exclude': []}
+                )
+
+            MetaExcludeID = schema_class_factory('MetaExcludeID')
+            MetaWithID = schema_class_factory('MetaWithID')
+
+            if hasattr(class_, 'created_at') and class_.created_at is not None:
+                MetaExcludeID.exclude.append('created_at')
+                MetaWithID.exclude.append('created_at')
+            if hasattr(class_, 'id') and class_.id is not None:
+                MetaExcludeID.exclude.append('id')
 
-            schema_class_name = '%s' % class_.__name__
+            schema_with_id_class_name = f'{class_.__name__}'
+            schema_no_id_class_name = f'{class_.__name__}NoID'
 
-            schema_class = type(
-                schema_class_name,
+            schema_with_id_class = type(
```

Comment:
This isn't exactly what I had in mind; if we change the utility function slightly, we would not need the class name specification on line 93, 94 or these type instantiations.  We can also have the factory insert the type into the namespace.

Reply:
What do you propose?

Reply:
A factory method that takes the name of the class and the schema and, perhaps, a list of excludes, and then does all the logic.

The main goal is to avoid the duplication that is currently there.

Reply:
I see what you mean. Sure, will do!

Reply:
@stefanv done!